### PR TITLE
[GitHub] Add codeowners for mlir/spirv

### PR DESCRIPTION
**Pull request by kuhar**
Add @antiagainst and @kuhar as codeowners for SPIR-V in MLIR. This is to assign reviewers automatically.

We already have a self-serve mechanism for subscribing to mlir/spirv issues (@llvm/pr-subscribers-mlir-spirv), but this codeowners change reflects a narrower set of core authors/maintainers for this part of the mlir.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -32,3 +32,12 @@
...
 /lldb/ @JDevlieghere
+
+# SPIR-V in MLIR.
+/mlir/include/mlir/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/lib/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/include/mlir/Conversion/SPIRVTo*/ @antiagainst @kuhar
+/mlir/include/mlir/Conversion/*ToSPIRV/ @antiagainst @kuhar
+/mlir/lib/Conversion/SPIRVTo*/ @antiagainst @kuhar
+/mlir/lib/Conversion/*ToSPIRV/ @antiagainst @kuhar
+/mlir/tools/mlir-tblgen/SPIRVUtilsGen.cpp @antiagainst @kuhar
```
Comment (antiagainst) [inline .github/CODEOWNERS:43]: We should also include paths in `/mlir/test` and `/mlir/docs` and `/mlir/unittests`? There are also `/mlir/include/mlir/Target/SPIRV` and `/mlir/lib/Target/SPIRV`. There might also be something else missed; would be good to do `find . -name "*SPIRV*" ` to check out.

**Review by antiagainst — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (kuhar) [reply .github/CODEOWNERS:49]: Done

**Review by antiagainst — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -32,3 +32,18 @@
...
+/mlir/include/mlir/Conversion/*ToSPIRV/ @antiagainst @kuhar
+/mlir/lib/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/lib/Target/SPIRV/ @antiagainst @kuhar
+/mlir/lib/Conversion/SPIRVTo*/ @antiagainst @kuhar
+/mlir/lib/Conversion/*ToSPIRV/ @antiagainst @kuhar
+/mlir/test/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/test/Target/SPIRV/ @antiagainst @kuhar
+/mlir/test/lib/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/unittests/Dialect/SPIRV/ @antiagainst @kuhar
+/mlir/tools/mlir-tblgen/SPIRVUtilsGen.cpp @antiagainst @kuhar
```
Comment (joker-eph) [inline .github/CODEOWNERS:49]: Can you use a wildcard? `/mlir/**/*SPIRV* ?

Comment (kuhar) [reply .github/CODEOWNERS:49]: I didn't know if codeowners supported recursive wildcards -- the documentation seems quite sparse to me: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax:
> A CODEOWNERS file uses a pattern that follows most of the same rules used in [gitignore](https://git-scm.com/docs/gitignore#_pattern_format) files.

and I'm not sure how to test it best. I will try it tomorrow.

Comment (joker-eph) [reply .github/CODEOWNERS:49]: In the past I tested it locally by modifying my .gitignore file and checking what git was saying when I modified some paths :)

Comment (kuhar) [reply .github/CODEOWNERS:49]: I checked with gitignore and it does work. Thanks! https://github.com/llvm/llvm-project/pull/72485

**Pull request merged**